modules/livestore/live_store.go

```diff
@@ -52,8 +51,9 @@ type LiveStore struct {
 	ingestPartitionID         int32
 	ingestPartitionLifecycler *ring.PartitionInstanceLifecycler
 
-	client  *kgo.Client
-	decoder *ingest.Decoder
+	client        KafkaClient
+	clientFactory KafkaClientFunc
```

Comment:
oof, generally would prefer not, but understand if it aids testing.

Reply:
I could probably get rid of the javaesque ClientFunc, but it needs some items from the constructor. I am more than happy to revisit this once we get it rolled out and refined a bit more.

Reply:
no that's fine. i think we can/should remove it from the struct but the factory pattern is nbd